hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/FileCountBySizeKeyCodec.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.spi.impl;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Longs;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.hadoop.hdds.utils.db.Codec;
+import org.apache.hadoop.ozone.recon.tasks.FileSizeCountKey;
+
+/**
+ * Codec to serialize/deserialize {@link FileSizeCountKey}.
+ */
+public final class FileCountBySizeKeyCodec implements Codec<FileSizeCountKey> {
+
+  private static final String KEY_DELIMITER = "_";
+
+  private static final Codec<FileSizeCountKey> INSTANCE = new FileCountBySizeKeyCodec();
+
+  public static Codec<FileSizeCountKey> get() {
+    return INSTANCE;
+  }
+
+  private FileCountBySizeKeyCodec() {
+    // singleton
+  }
+
+  @Override
+  public Class<FileSizeCountKey> getTypeClass() {
+    return FileSizeCountKey.class;
+  }
+
+  @Override
+  public byte[] toPersistedFormat(FileSizeCountKey key) {
+    Preconditions.checkNotNull(key, "Null object can't be converted to byte array.");
+    
+    // Serialize: volume + delimiter + bucket + delimiter + fileSize (8 bytes)
+    byte[] volumeBytes = key.getVolume().getBytes(UTF_8);
```

Comment:
we can use Message format with splitter or String builder here. Underscore is used as splitter, what if this is part of volume or bucket name ? can we use same splitter "/" for volume and bucekt and others ?